### DISCUSSION TITLE
Transitive Dependency issue in Pygments used by HTTPX as used by Notion Client

### DISCUSSION THREAD

**Discussion by iPoetDev** [ramnes/notion-sdk-py#191] [General]
FYI: Info

I ran into a transitive dependency issue for a package called pygments

![image](https://user-images.githubusercontent.com/51715025/234050120-3c736c13-1784-425a-980d-bfabe55f8f9f.png)

and have logged a ticket with pygments for version 2.15.1 (I think) and with HTTPX >= 0.24.0

https://github.com/pygments/pygments/issues/2418

So you may want to update these as issues are patches or requirements are sync across the dependency tree

**Comment by ramnes** [comment]
Hey, thanks for reaching out but:
1. We do not pin dependencies, so the CI always runs against the latest versions of the dependency tree – and currently passes.
2. That also means that a `pip install notion-client` should get you the latest version for httpx. 
3. Pygments is not a dependency for notion-client.

Are you sure the problem is not related with your setup? If so, what are you proposing exactly?

**Reply by iPoetDev** [reply]
Pygments is a dependency of httpx

I checked with pygments crew, their pygments is passing and the issue is not on their end (as this error is removed/source control is removed)

I logged a discussion on encode/httpx (is that the right one) and tagged https://github.com/encode/httpx/issues/2677

I don't know, it was first run/first attempt at this after fresh install (using requirements.txt on pycharm 2023.01 Edu) so this was first touch.

As a starter, not a newbie, I can resolve locally ... but it is my early stage in using python? What could I have missed in my setup that would have impacted this ? 
- 3.11
-  venv activated
- PyCharm 2023.01 Edu
- Requirements.txt with latest

![image](https://user-images.githubusercontent.com/51715025/234365080-f99710c2-40a9-4470-8f0e-09e8af3f55cb.png)

**Reply by iPoetDev** [reply]
Pygment's issue
https://github.com/pygments/pygments/issues/2418

**Reply by iPoetDev** [reply]
Unfortunately, if I can move beyond this blocker – I will have to revert to Gsheets for my assignment, rather than my preferred datastore (Notion)

**Reply by ramnes** [reply]
Pygment is an _optional_ dependency of httpx, that's only fetched when `httpx[cli]` is asked, which we don't, so I repeat: pygments is not a dependency for notion-client.

There are a lot of libraries in your `requirements.txt`, and most of them are not meant to work with notion-client (all the `notion-*`), why do you think the problem comes from notion-client in particular?

**Reply by ramnes** [reply]
I just installed all your libraries except pygment, and pygment isn't even pulled:

```
$ pip install google gspread google-auth google-auth-oauthlib notion-cli-py 'notion-client>=2.0.0' notion-database notion-oauth-handler notion-toolkit 'notion>=0.0.27' notion-python python-dotenv 'certifi>=2022.12.7' oauthlib 'requests>=2.28.2' requests-oauthlib 'urllib3>=1.26.15' 'httpx>=0.24.0' yarl 'click>=8.1.3' fire
...
$ pipdeptree | grep -i pygment
$
```

So I guess you just have to uninstall pygment.
